### DISCUSSION TITLE
Is there any way to remove some URL query before SSR

### DISCUSSION THREAD

**Discussion by suningkooboo** [nuxt/nuxt#23142] [Questions]
Is there any way to remove tracking URL queries like utm_source, utm_campaign before SSR render, especially before routeRules cache handling so the cache key not include these queries.   

I try to do it in Nitro server/middleware.  But I didn't find a good API to do this. If I use navigateTo or sendRedirect, it will cause page jump. We only want to **ignore** such queries to normalize request URL in SSR, but keep it work in client-side.

**Comment by sauravbv** [comment]
Following, using a proxy server to add query params and hit the nuxt server, but the nuxt server is returning with query params in the SSR response. This is happening after `Nuxt 3.6.0` upgrade.

**Reply by Vi-jay** [reply]
It is now 3.8. How can I get this feature?

**Reply by sauravbv** [reply]
@Vi-jay  Answered here

https://github.com/nuxt/nuxt/issues/23153

**Comment by coltisor** [comment]
@suningkooboo you can check my workaround here: https://github.com/nuxt/nuxt/issues/33728
